pkg/core/native/native_neo.go

```diff
@@ -283,13 +288,68 @@ func (n *NEO) OnPersist(ic *interop.Context) error {
 func (n *NEO) PostPersist(ic *interop.Context) error {
 	gas := n.GetGASPerBlock(ic.DAO, ic.Block.Index)
 	pubs := n.GetCommitteeMembers()
-	index := int(ic.Block.Index) % len(ic.Chain.GetConfig().StandbyCommittee)
-	gas.Mul(gas, big.NewInt(committeeRewardRatio))
-	n.GAS.mint(ic, pubs[index].GetScriptHash(), gas.Div(gas, big.NewInt(100)))
+	committeeSize := len(ic.Chain.GetConfig().StandbyCommittee)
+	index := int(ic.Block.Index) % committeeSize
+	committeeReward := new(big.Int).Mul(gas, big.NewInt(committeeRewardRatio))
+	n.GAS.mint(ic, pubs[index].GetScriptHash(), committeeReward.Div(committeeReward, big.NewInt(100)))
+
+	if shouldUpdateCommittee(ic.Block.Index, ic.Chain) {
+		var voterReward = big.NewInt(voterRewardRatio)
+		voterReward.Mul(voterReward, gas)
+		voterReward.Mul(voterReward, big.NewInt(voterRewardFactor*int64(committeeSize)))
+		var validatorsCount = ic.Chain.GetConfig().ValidatorsCount
+		voterReward.Div(voterReward, big.NewInt(int64(committeeSize+validatorsCount)))
+		voterReward.Div(voterReward, big.NewInt(100))
+
+		var cs = n.committee.Load().(keysWithVotes)
+		var buf = make([]byte, 4)
+		var si = new(state.StorageItem)
+		var key = make([]byte, 38)
+		for i := range cs {
+			if cs[i].Votes.Sign() > 0 {
+				tmp := big.NewInt(1)
+				if i < validatorsCount {
+					tmp = big.NewInt(2)
+				}
+				tmp.Mul(tmp, voterReward)
+				tmp.Div(tmp, cs[i].Votes)
+
+				key = makeVoterKey([]byte(cs[i].Key), key[:34])
+				var reward = n.getGASPerVote(ic.DAO, key, ic.Block.Index+1)
+				tmp.Add(tmp, &reward[0])
+
+				binary.BigEndian.PutUint32(buf, ic.Block.Index+1)
+				key = append(key, buf...)
```

Comment:
`binary.BigEndian.PutUint32(key[34:], ic.Block.Index+1)` and drop the buf/append?

Reply:
fixed.